### PR TITLE
[OCONF-165] coconut conf show command

### PR DESCRIPTION
**Pull request by graduta**

### CODE REVIEW

File: coconut/configuration/configuration.go [hunk 1]
```diff
@@ -215,10 +277,65 @@ func formatListOutput( cmd *cobra.Command, output []string)(parsedOutput []byte,
...
+	if len(keys) == 0 {
+		err = errors.New(fmt.Sprintf("No keys found"))
+		return "", err, 3
+	}
+
+	var maxTimeStamp string
+	for _, key := range keys {
+		componentTimestamp := strings.TrimPrefix(key, keyPrefix + "/")
+		fmt.Println(componentTimestamp)
+		if strings.Compare(componentTimestamp, maxTimeStamp) > 0 {
```
Comment (teo) [inline coconut/configuration/configuration.go]: This (and similar situations, if any) should be an integer comparison, not a string comparison.

File: coconut/configuration/configuration.go [hunk 2]
```diff
@@ -48,10 +49,17 @@ var log = logger.New(logrus.StandardLogger(), "coconut")
 
 type RunFunc func(*cobra.Command, []string)
 
-type ConfigurationCall func(configuration.Source, *cobra.Command, []string, io.Writer) (error)
+type ConfigurationCall func(*configuration.ConsulSource, *cobra.Command, []string, io.Writer) (error, int)
 
 var componentsPath = "o2/components/"
 
+var InputRegex, _ = regexp.Compile(`^([a-zA-Z0-9-]+)(\/[a-z-A-Z0-9-]+)?(\@[0-9]+)?$$`)
```
[the remark this answers is not in the document]
Comment (graduta) [reply coconut/configuration/configuration.go]: 👍

File: coconut/configuration/configuration.go [hunk 3]
```diff
@@ -123,77 +131,131 @@ func Dump(cfg configuration.Source, cmd *cobra.Command, args []string, o io.Writ
...
+				params := strings.Split(arg, "/")
+				component = params[0]
+				entry = params[1]
+				timestamp = params[2]
+			} else if strings.Contains(args[0], "/") {
+				// coconut conf show component/entry
+				params := strings.Split(args[0], "/")
+				component = params[0]
+				entry = params[1]
+				}
```
[the remark this answers is not in the document]
Comment (graduta) [reply coconut/configuration/configuration.go]: 👍

File: coconut/configuration/configuration.go (hunk 1 above)
Comment (graduta) [reply coconut/configuration/configuration.go]: 👍

File: coconut/configuration/configuration.go [hunk 4]
```diff
@@ -73,15 +81,15 @@ func WrapCall(call ConfigurationCall) RunFunc {
 			log.WithPrefix(cmd.Use).
 				WithFields(fields).
 				Fatal("cannot query endpoint")
-			os.Exit(1)
+			os.Exit(2)
```
[the remark this answers is not in the document]
Comment (graduta) [reply coconut/configuration/configuration.go]: 👍

File: coconut/configuration/configuration.go [hunk 5]
```diff
@@ -53,12 +54,15 @@ type ConfigurationCall func(*configuration.ConsulSource, *cobra.Command, []strin
 
 var componentsPath = "o2/components/"
 
-var InputRegex, _ = regexp.Compile(`^([a-zA-Z0-9-]+)(\/[a-z-A-Z0-9-]+)?(\@[0-9]+)?$$`)
+var InputRegex, __ = regexp.Compile(`^([a-zA-Z0-9-]+)(\/[a-z-A-Z0-9-]+){1}(\@[0-9]+)?$`)
```
Comment (teo) [inline coconut/configuration/configuration.go]: `regex.MustCompile` instead of ignoring error please.

Comment (graduta) [reply coconut/configuration/configuration.go]: Argh, bad reverting. Updated now

**Pull request merged**